### PR TITLE
Shared app context

### PR DESCRIPTION
**Pull request by rachaelparris**

### CODE REVIEW

File: shared/src/business/useCases/courtIssuedOrder/updateCourtIssuedOrderInteractor.test.js [hunk 1]
```diff
@@ -83,29 +91,19 @@ describe('updateCourtIssuedOrderInteractor', () => {
   });
```
Comment (rachelschneiderman) [inline shared/src/business/useCases/courtIssuedOrder/updateCourtIssuedOrderInteractor.test.js:87]: Nit: we can refactor this to take advantage of .rejects.toThrow() instead.

`await expect(
      uploadExternalDocumentsInteractor({
        applicationContext
      }),
    ).rejects.toThrow('Unauthorized');`

File: web-client/src/presenter/actions/AdvancedSearch/validateCaseAdvancedSearchAction.test.js [hunk 2]
```diff
@@ -1,31 +1,29 @@
...
-
     presenter.providers.path = {
       error: errorStub,
       success: successStub,
     };
   });
 
   it('validates advanced case search successfully', async () => {
-    validateCaseAdvancedSearchStub = jest.fn().mockReturnValue({});
+    applicationContext.getUseCases().validateCaseAdvancedSearchInteractor = jest
```
Comment (rachelschneiderman) [inline web-client/src/presenter/actions/AdvancedSearch/validateCaseAdvancedSearchAction.test.js]: Nit: I don't think we have to create a whole new jest function here, we can just tack on the .mockReturnValue() :D

File: web-client/src/presenter/actions/caseAssociation/validateAddIrsPractitionerAction.test.js [hunk 3]
```diff
@@ -1,37 +1,35 @@
...
       user: { userId: 'abc' },
     };
 
-    presenter.providers.applicationContext = {
-      getUseCases: () => ({
-        validateAddIrsPractitionerInteractor: validateAddIrsPractitionerInteractorStub,
-      }),
-    };
-
+    presenter.providers.applicationContext = applicationContext;
```
Comment (rachelschneiderman) [inline web-client/src/presenter/actions/caseAssociation/validateAddIrsPractitionerAction.test.js:21]: Nit: can this and line 13 be combined?

File: web-client/src/presenter/actions/AdvancedSearch/validateCaseAdvancedSearchAction.test.js (hunk 2 above)
Comment (rachaelparris) [reply web-client/src/presenter/actions/AdvancedSearch/validateCaseAdvancedSearchAction.test.js]: I think I did this several times. I'll fix those!

**Review by rachelschneiderman — APPROVED** [review]

File: web-client/src/presenter/actions/caseAssociation/validateAddIrsPractitionerAction.test.js (hunk 3 above)
Comment (sutt0n) [reply web-client/src/presenter/actions/caseAssociation/validateAddIrsPractitionerAction.test.js:21]: i'll commit this for the next merge

Comment (rachelschneiderman) [reply web-client/src/presenter/actions/caseAssociation/validateAddIrsPractitionerAction.test.js:21]: Oops sorry, Rachael and I talked about this in slack.

We decided that even though it's a little wordy maybe it's better to leave it as is. If you set it to '= applicationContextForClient', then later in the tests when you’re accessing it, you have to write 'applicationContextForClient.getUseCases()' which may be confusing.

Comment (rachaelparris) [reply web-client/src/presenter/actions/caseAssociation/validateAddIrsPractitionerAction.test.js:21]: I'm shortening it to just `presenter.providers.applicationContext = applicationContextForClient;` in tests where we don't later access the applicationContext.

**Pull request merged**